### PR TITLE
Rosetta: Adds nonce to the /account/balance response

### PR DESCRIPTION
**Pull request by bkase**
Explain your changes:
* Adds nonce to the /account/balance response via archive DB query change

Explain how you tested your changes:
* Manually connected to mainnet nodes and tried queries -- the correct nonce is retrieved on non-historical queries and what looks to be correct nonces are returned on historical ones.

### CODE REVIEW

**Comment by psteckler** [comment]
Per earlier conversation:
- a comment explaining the strategy, including the use of `NULL`s and `MIN`, would help future readers
- handle case where there are no user commands for a key
- distinguish two `pending_chain` blocks, for example, `pending_chain_balance` and `pending_chain_nonce`

**Comment by psteckler** [comment]
For the `pending` query, I think you may be able to compute the `pending_chain`, and then the `full_chain` just once, and then `SELECT` what you need from that.

File: src/app/rosetta/lib/account.ml [hunk 1]
```diff
@@ -69,33 +80,104 @@ module Sql = struct
...
+              INNER JOIN balances             bal  ON full_chain.id = bal.block_id
+              INNER JOIN public_keys          pks  ON bal.public_key_id = pks.id
 
               WHERE pks.value = $1
               AND full_chain.height <= $2
 
               ORDER BY (bal.block_height, bal.block_sequence_no, bal.block_secondary_sequence_no) DESC
               LIMIT 1
+            )
+            UNION ALL
```
Comment (psteckler) [inline src/app/rosetta/lib/account.ml:92]: think the `UNION` can be flush-left, it's at same level as the `WITH RECURSIVE`

**Comment by bkase** [comment]
> For the pending query, I think you may be able to compute the pending_chain, and then the full_chain just once, and then SELECT what you need from that.

I'll make an issue for this and TODO in the a comment and we can follow up with it. I played around a bit, but couldn't get it to work and it's not necessary for now I think

**Review by psteckler — APPROVED** [review]

**Comment by joaosreis** [comment]
Closed as nonce is already being retrieved in some other way in the current main branches.

**Pull request closed without merging**